### PR TITLE
[SRVKE-1324] Avoid reconciliation loops

### PR DESCRIPTION
**Pull request by pierDipi**
This is a  fix for SRVKE-1324

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

## Proposed Changes

- set creation timestamp to 0 on every resource
- global resync only when some config maps change (config-tracing and kafka-config-logging)
- remove match expression from webhook definition since it is changed by Knative webhook logic

### CODE REVIEW

File: knative-operator/pkg/controller/knativekafka/knativekafka_controller.go [hunk 1]
```diff
@@ -681,3 +683,14 @@ func executeStages(instance *serverlessoperatorv1alpha1.KnativeKafka, manifest *
 	}
 	return nil
 }
+
+func removeCreationTimestamp(manifest *mf.Manifest, _ *serverlessoperatorv1alpha1.KnativeKafka) error {
+	// Avoid multiple unnecessary resource updates and reconciler looping
+	// due to creationTimestamp set to `null` on some resources.
```
Comment (skonto) [inline knative-operator/pkg/controller/knativekafka/knativekafka_controller.go:689]: When `null` is set? Could you use a predicate to filter these changes on meta instead?

Comment (pierDipi) [reply knative-operator/pkg/controller/knativekafka/knativekafka_controller.go:698]: I'm using the same approach the other functions are using, see 
https://github.com/openshift-knative/serverless-operator/blob/26a310ee58a38e70d3f7bd1d1d057891d06e287f/knative-operator/pkg/controller/knativekafka/images.go#L86-L88
(there are many other instances of the same logic)

**Review by matzew — APPROVED** [review]
/lgtm
/approve

File: knative-operator/deploy/resources/knativekafka/controller/eventing-kafka-controller.yaml [hunk 2]
```diff
@@ -1648,7 +1648,6 @@ webhooks:
     reinvocationPolicy: IfNeeded
     matchPolicy: Equivalent
     namespaceSelector:
-      matchExpressions: [ ]
```
Comment (matzew) [inline knative-operator/deploy/resources/knativekafka/controller/eventing-kafka-controller.yaml:1651]: upstream removes this here: https://github.com/knative-sandbox/eventing-kafka-broker/pull/2715

**Pull request merged**